src/CloudFrontClientFactory.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Madmatt\SilverStripeS3;
+
+use Aws\CloudFront\CloudFrontClient;
+use SilverStripe\Core\Injector\Factory;
+
+class CloudFrontClientFactory extends ClientFactory implements Factory
```

Comment:
I think it might be good to merge `CloudFrontClientFactory` and `S3ClientFactory` into one factory class and then use $service as a switch (from what I can see $service contains the name of the class to be created). This would eliminate the inheritance and remove the amount of classes needed from 3 to 1.